### ISSUE TITLE
Add Windows Agent with Docker in Linux container mode.

### ISSUE THREAD

**Issue by joshmackey** [microsoft/azure-pipelines-image-generation#702]
**Tool infomation**
- Tool name: Docker with Linux Containers (On Windows)
- Add or update? Add
- If this is an add request:
  Looking to have windows agents that can run linux docker containers.

**Areas and ecosystems**

- [ ] Android
- [ ] C / C++
- [X ] Container tools
- [ ] Database servers or clients
- [ ] General purpose tools
- [ ] Go
- [ ] Java
- [ ] Javascript, Node.js, and TypeScript
- [ ] iOS, macOS, and Apple-specific tools
- [X ] .NET Core or .NET Desktop
- [ ] PHP
- [ ] Python
- [ ] Ruby
- [ ] Rust
- [ ] Scala
- [X] Testing and code coverage
- [ ] Xamarin
- [ ] Other

**Image impacted**
Windows

Basically, we have a bunch of integration tests that we need to run using the VSTest task (Windows only) that have dependencies on services in which only linux docker containers exist. The regular `dotnet test` linux runner doesn't have the functionality we need yet (Slicing, mostly).

**Comment by joshmackey** [comment]
Any activity on this?

**Comment by leomsft** [comment]
In our case, we have dependency on both Linux and Windows containers in the same pipeline job. Given that Linux agent can't host Windows container, having LCOW support on hosted Windows agent will be great.

**Comment by ChrisCAdsk** [comment]
We have the same issue, we start servers using docker images only available for Linux.
The community edition docker for Windows comes with a tool `DockerCli.exe` which can switch between LCOW and Windows containers.

**Comment by SabotageAndi** [comment]
We also have this issue to test our software in Docker containers.

**Comment by MarkusHorstmann** [comment]
We are building Linux docker images from a SLN that contains all out projects (mix of .Net Framework 3.5, 4.5, 4.7, .Net Core). We currently have to use an external Linux docker, which is cumbersome (nasty build pipeline initialization tasks etc.) and expensive. 

It would be great if we could build everything on a single Windows hosted agents, as that is how all our Devs also build.

**Comment by syedhassaanahmed** [comment]
+1

**Comment by kaylangan** [comment]
I've copied this issue over to the actions/virtual-environments repo where development has been moved to. Please follow the issue there: https://github.com/actions/virtual-environments/issues/252
